src/components/web-monitoring-ui.jsx

```diff
@@ -103,6 +103,8 @@ export default class WebMonitoringUi extends React.Component {
           return api.getPages(query);
         }
       })
+      // Set state.pages = error; downstream code should check `pages` type.
+      .catch(error => error)
```

Comment:
I am very much not-in-love with this pattern that we’ve slowly started using, but fixing requires some re-thinking of how we move data around.

Reply:
Agreed, I know #126 (Better state management with redux) has been in the backlog for awhile. It pains me as well to not have had the resources to tackle these problems. 😓